src/app/components/statistic-page/statistic-page.component.ts

```diff
@@ -14,31 +14,42 @@ export class StatisticPageComponent implements AfterViewInit {
 
   public showStatisticTable = false;
   public toggleBtnText = 'Table';
+  public isStatisticDataExist = false;
 
-  public quizzesPlayedData: ChartData<'pie'>;
-  public correctAnswersData: ChartData<'pie'>;
-  public averagePointsData: ChartData<'pie'>;
-  public averageTimeData: ChartData<'pie'>;
+  public quizzesPlayedData!: ChartData<'pie'>;
+  public correctAnswersData!: ChartData<'pie'>;
+  public averagePointsData!: ChartData<'pie'>;
+  public averageTimeData!: ChartData<'pie'>;
   public pieChartType: ChartType = 'pie';
 
   @ViewChild(MatSort) sort!: MatSort;
-  public displayedColumns: string[];
-  public dataSource: MatTableDataSource<StatisticTableData>;
-  private _statisticTableData: StatisticTableData[];
-
-  constructor(private statisticService: StatisticService) {
-    this.quizzesPlayedData = this.statisticService.getStatisticData('quizzesPlayed');
-    this.correctAnswersData = this.statisticService.getStatisticData('correctAnswers');
-    this.averagePointsData = this.statisticService.getStatisticData('averagePoints');
-    this.averageTimeData = this.statisticService.getStatisticData('averageTime');
-
-    this._statisticTableData = this.statisticService.getStatisticTableData();
-    this.displayedColumns = Object.keys(this._statisticTableData[0]);
-    this.dataSource = new MatTableDataSource(this._statisticTableData);
+  public displayedColumns!: string[];
+  public dataSource!: MatTableDataSource<StatisticData>;
+
+  private _statisticData: StatisticData[];
+
+  constructor(public statisticService: StatisticService) {
+    this._statisticData = this.statisticService.getCurrentStatisticData();
+  }
+
+  ngOnInit(): void {
+    if (this._statisticData.length) {
+      this.isStatisticDataExist = true;
+      this.quizzesPlayedData = this.statisticService.getStatisticChartData('quizzesPlayed');
```

Comment:
Create a const with this function - this.statisticService.getStatisticChartData(), then access it in following variables. Much cleaner code.